pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -745,58 +895,150 @@ def compute_eigvals(*params: TensorLike, **hyperparams) -> TensorLike:
+
+        A detailed description of the algorithm and integer decomposition subroutine
+        is provided in ``pennylane/ops/qubit/pcphase_decomposition.md``. In the following
+        we provide a detailed example for illustration purposes.
+
+        **Detailed example**
+
+        Consider the projector-controlled phase gate on :math:`n=4` qubits and with
+        :math:`d=\texttt{dim}=3`, i.e,
+
+        >>> op_3 = qml.PCPhase(1.23, dim=3, wires=[0, 1, 2, 3])
+
+        It acts on :math:`N=2^n=16`-dimensional vectors and is described by
+
+        .. math:: \Pi(\phi) = \exp(i\phi G) = \exp(i\phi(2\Pi-\mathbb{I}_N)),
+
+        where :math:`G` is a diagonal matrix with :math:`d=3` ones, followed by
+        :math:`2^n-d = 16 - 3=13` negative ones. Accordingly, :math:`\Pi` is diagonal with
+        :math:`3` ones and :math:`13` zeros.
+
+        First, we implement the global phase generated by :math:`\mathbb{I}_N` with
+        a :class:`~.GlobalPhase` gate with angle :math:`-\phi`.
+        Then we decompose :math:`d` into powers of two with positive or negative sign, via
+        :math:`d=3=4-1 = 2^2-2^0`. This decomposition tells us that we can write the
+        target gate with two (multi-)controlled phase shift gates. For this, we rewrite
+        the projector :math:`\Pi` according to the decomposition as
+
+        .. math::
+
+            \Pi &= \text{diag}(1, 1, 1, 0, 0, \dots, 0)\\
+            &=\text{diag}(1, 1, 1, 1, 0, \dots, 0)
+            -\text{diag}(0, 0, 0, 1, 0, \dots, 0)
+
+        where :math:`0,\dots, 0` indicates :math:`12` zeros each time.
+        How do we realize this projector decomposition on the gate level?
+
+        A singly-controlled phase shift gate applies a phase to a quarter of all computational
+        basis states (the control filters by the state of one qubit, and the phase shift gate
+        itself filters by the :math:`|1\rangle` state of the target qubit, cutting the number
+        of states we are acting on in half each time).
+        For :math:`n=4`, this amounts to :math:`2^4/4=4` states, which is exactly
+        what we need for the first term above. To apply the phase to the *first* four states,
+        :math:`|0000\rangle`, :math:`|0001\rangle`, :math:`|0010\rangle`, and :math:`|0011\rangle`,
+        we want to "filter by" the first two qubits being in the :math:`|0\rangle` state.
+        For qubit :math:`0`, we do this by controlling on the :math:`|0\rangle` state.
+        For qubit :math:`1`, we pick it as the target of the controlled phase shift operation.
+        Generically, this would make it act on the :math:`|1\rangle` state, so we simply flip
+        qubit :math:`1` before and after the operation to apply the phase to the :math:`|0\rangle`
+        state instead.
+        Thus, we conclude this first step by applying the gates
+        ``qml.X(1)``, ``qml.ctrl(qml.PhaseShift(2 * phi, 1), control=[0], control_values=[0])``,
+        and ``qml.X(1)``.
```

Comment:
this helps a looooot! thanks for the update!

Reply:
Nice, very happy to hear that!